### PR TITLE
Converting Azd ToDo ARM templates -> ADE compatible

### PR DESCRIPTION
**Pull request by v-jiaodi**
There are three EnvironmentDefinitions have been implemented, ADE compatible ARM templates as follow:

1. App Service with Cosmos 
	• todo-python-mongo
	• todo-java-mongo
	• todo-nodejs-mongo
2. Container App with Cosmos
	• todo-python-mongo-aca
	• todo-nodejs-mongo-aca
	• todo-java-mongo-aca
3. Function App / SWA with Cosmos 
	• todo-nodejs-mongo-swa-func
	• todo-python-mongo-swa-func

@rajeshkamal5050 and @wbreza for notification.

### CODE REVIEW

File: EnvironmentDefinitions/Terraform/App_Service_with_Cosmos_using_TF/main.tf
Comment (wbreza) [inline EnvironmentDefinitions/Terraform/App_Service_with_Cosmos_using_TF/main.tf:1]: Please split terraform templates into their own PR.

File: EnvironmentDefinitions/ARMTemplates/App_Service_with_Cosmos/environment.yaml
Comment (wbreza) [inline EnvironmentDefinitions/ARMTemplates/App_Service_with_Cosmos/environment.yaml:15]: I understand this param is specific to the `azd` scenario but we should also make this environment definition user friendly for people not using `azd`.

Perhaps we should add empty string as an allowed value and also introduce a parameter for target language.  This would then prompt the user to select a language that can be reused whether or not they are using from within `azd`.

This same comment applies to all the ARM templates

File: EnvironmentDefinitions/ARMTemplates/App_Service_with_Cosmos/azuredeploy.json
Comment (wbreza) [inline EnvironmentDefinitions/ARMTemplates/App_Service_with_Cosmos/azuredeploy.json:16]: The API app service does not appear to be coming online for me. `azd up` command completed successfully.

Comment (wbreza) [reply EnvironmentDefinitions/ARMTemplates/App_Service_with_Cosmos/azuredeploy.json:16]: I selected the node template, but after deployment discovered the runtime stack was set to Java.

File: EnvironmentDefinitions/ARMTemplates/App_Service_with_Cosmos/azuredeploy.json
Comment (wbreza) [inline EnvironmentDefinitions/ARMTemplates/App_Service_with_Cosmos/azuredeploy.json:237]: Looks like we're missing the Node use case here.

File: EnvironmentDefinitions/ARMTemplates/Function_App_with_Cosmos/azuredeploy.json
Comment (wbreza) [inline EnvironmentDefinitions/ARMTemplates/Function_App_with_Cosmos/azuredeploy.json:2]: Function app API is not coming online.  Selected python in this scenario.

**Review by wbreza — COMMENTED** [review]
Overall looks good.

The `EnvironmentDefinitions` folder changed name to `Environment-Definitions`. Please rebase from main and address.

`azd provision/up` works fine, but having issues with apps coming online.  I believe it has to do with the specific runtime settings being applied.

Also added feedback around the language getting bubbled up to a root level param in addition to the repo url.

**Comment by v-jiaodi** [comment]
@wbreza Based on your comments, I have updated the code. Please review.

File: Environment-Definitions/ARMTemplates/App_Service_with_Cosmos/environment.yaml
Comment (wbreza) [inline Environment-Definitions/ARMTemplates/App_Service_with_Cosmos/environment.yaml:11]: Please provide a more user friendly display name like `Runtime Type`
```suggestion
    name: Runtime Type
```

File: Environment-Definitions/ARMTemplates/App_Service_with_Cosmos/environment.yaml
Comment (wbreza) [inline Environment-Definitions/ARMTemplates/App_Service_with_Cosmos/environment.yaml:12]:
```suggestion
    description: 'Runtime type of the App Service runtime,eg python, nodejs or java.'
```

File: Environment-Definitions/ARMTemplates/Container_App_with_Cosmos/environment.yaml
Comment (wbreza) [inline Environment-Definitions/ARMTemplates/Container_App_with_Cosmos/environment.yaml:11]:
```suggestion
    name: Runtime Type
```

File: Environment-Definitions/ARMTemplates/Container_App_with_Cosmos/environment.yaml
Comment (wbreza) [inline Environment-Definitions/ARMTemplates/Container_App_with_Cosmos/environment.yaml:12]:
```suggestion
    description: 'Runtime type of the container App runtime, eg python, nodejs or java.'
```

File: Environment-Definitions/ARMTemplates/Function_App_with_Cosmos/environment.yaml
Comment (wbreza) [inline Environment-Definitions/ARMTemplates/Function_App_with_Cosmos/environment.yaml:11]:
```suggestion
    name: Runtime Type
```

File: Environment-Definitions/ARMTemplates/Function_App_with_Cosmos/environment.yaml
Comment (wbreza) [inline Environment-Definitions/ARMTemplates/Function_App_with_Cosmos/environment.yaml:12]:
```suggestion
    description: 'Runtime type of the Function App runtime, eg python, nodejs or java.'
```

**Review by wbreza — COMMENTED** [review]
A few minor comments around the user friendly display name and descriptions but is looking good.  I'll be performing a few more manual tests before final approval.

File: Environment-Definitions/ARMTemplates/App_Service_with_Cosmos/azuredeploy.json
Comment (wbreza) [inline Environment-Definitions/ARMTemplates/App_Service_with_Cosmos/azuredeploy.json:1]: Getting application error in Azure after deploying.

### Repro Steps

Prereqs:
- Setup devcenter with this as a catalog

Steps:
1. `azd init` in empty folder
2. Select `App Service With Cosmos`
3. Enter valid environment name when prompted
4. `azd up` - Select `python` as runtime type when prompted
5. Visit web frontend of deployed app, app should be working

Expected:
- Visiting frontend web app and app should be fully functional

Actual: 
- Getting generic `Application Error` message

image

File: Environment-Definitions/ARMTemplates/Container_App_with_Cosmos/environment.yaml
Comment (wbreza) [inline Environment-Definitions/ARMTemplates/Container_App_with_Cosmos/environment.yaml:2]:
```suggestion
name: Container-App-with-Cosmos
```

File: Environment-Definitions/ARMTemplates/App_Service_with_Cosmos/azuredeploy.json
Comment (wbreza) [reply Environment-Definitions/ARMTemplates/App_Service_with_Cosmos/azuredeploy.json:1]: FYI - it looks like it eventually came on line just took much longer than expected...

Comment (wbreza) [reply Environment-Definitions/ARMTemplates/App_Service_with_Cosmos/azuredeploy.json:1]: https://app-web-sbij2o6nsyu5i.azurewebsites.net/

Comment (v-jiaodi) [reply Environment-Definitions/ARMTemplates/App_Service_with_Cosmos/azuredeploy.json:1]: After successful deployment, it takes a few minutes for the program to be successfully launched. This issue occasionally occurs during our testing, which is related to the environment, network, and other aspects of the program deployment.
I will verify this issue again.

Comment (v-jiaodi) [reply Environment-Definitions/ARMTemplates/App_Service_with_Cosmos/azuredeploy.json:1]: @wbreza Regarding this issue, through investigation, we found that this issue actually occurs when opening the app link after creating the resource. After the code is deployed, it can run normally after waiting for a few minutes.

For `todo-python-mongo`, execute `azd provision` with `main.bicep` to create resources, and open the app link to encounter the same error.
The root cause is that `appCommandLine` was executed in the app service before deploying the code.

**Comment by v-jiaodi** [comment]
@wbreza The above comments have been resolved. Please review.

**Review by wbreza — APPROVED** [review]

**Comment by wbreza** [comment]
@jacksonkays - Can you please review.

File: Environment-Definitions/ARMTemplates/App_Service_with_Cosmos/environment.yaml
Comment (jacksonkays) [inline Environment-Definitions/ARMTemplates/App_Service_with_Cosmos/environment.yaml:20]:
```suggestion
    name: Repo URL
```

File: Environment-Definitions/ARMTemplates/Container_App_with_Cosmos/environment.yaml
Comment (jacksonkays) [inline Environment-Definitions/ARMTemplates/Container_App_with_Cosmos/environment.yaml:20]:
```suggestion
    name: Repository URL
```

File: Environment-Definitions/ARMTemplates/Function_App_with_Cosmos/environment.yaml
Comment (jacksonkays) [inline Environment-Definitions/ARMTemplates/Function_App_with_Cosmos/environment.yaml:19]:
```suggestion
    name: Repository URL
```

File: Environment-Definitions/ARMTemplates/App_Service_with_Cosmos/environment.yaml
Comment (jacksonkays) [inline Environment-Definitions/ARMTemplates/App_Service_with_Cosmos/environment.yaml:2]: We might want to change these environment definition names to having a connection to AZD such as 'App-Service-With-Cosmos-AZD' or 'AZD-App-Service-With-Cosmos' so they can be seen when accessing them in AZD CLI or DevPortal. I'll leave this up to Sagar though.

Comment (sagarlankala) [reply Environment-Definitions/ARMTemplates/App_Service_with_Cosmos/environment.yaml:2]: Below is my suggestion

1. App-Service-with-Cosmos_AZD-template
2. Container-App-with-Cosmos_AZD-template
3. Function-App-with-Cosmos_AZD-template

**Comment by v-jiaodi** [comment]
@jacksonkays I have updated environment definition name based on Sagar's suggestions, please review.

**Review by jacksonkays — APPROVED** [review]

**Pull request merged**
